### PR TITLE
feat: support "use" in GRPC requests

### PR DESCRIPTION
**Pull request by MichaelScofield**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

This PR:

- supports "use catalog and schema" (behaved like the "use" in MySQL) in GRPC requests
- refactor the codes, instead of using the `DEFAULT_CATALOG_NAME` and `DEFAULT_SCHEMA_name` consts everywhere, try to use the catalog and schema in query context
- use a seperate schema for each sqlness test case
  - act as tests for this feature, since sqlness is using GRPC to request the database
  - table names are made short and concise again 
- make Meta use memory kvstore instead of etcd when running sqlness test for distribute mode
- prepare for implementing "use" in distribute mode (tracked in #670 )

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [x]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

**Comment by killme2008** [comment]
@MichaelScofield There are some conflicts.

**Comment by MichaelScofield** [comment]
@killme2008 resolved, PTAL

File: src/mito/src/engine.rs [hunk 1]
```diff
@@ -489,8 +488,8 @@ impl<S: StorageEngine> MitoEngineInner<S> {
     }
 
     async fn alter_table(&self, _ctx: &EngineContext, req: AlterTableRequest) -> Result<TableRef> {
-        let catalog_name = req.catalog_name.as_deref().unwrap_or(DEFAULT_CATALOG_NAME);
-        let schema_name = req.schema_name.as_deref().unwrap_or(DEFAULT_SCHEMA_NAME);
+        let catalog_name = &req.catalog_name;
+        let schema_name = &req.schema_name;
         let table_name = &req.table_name.clone();
```
Comment (fengjiachun) [inline src/mito/src/engine.rs:493]: Can avoid to `clone`?

File: src/session/src/context.rs [hunk 2]
```diff
@@ -32,46 +34,55 @@ impl Default for QueryContext {
...
-            current_catalog: ArcSwapOption::new(Some(Arc::new(catalog))),
-            current_schema: ArcSwapOption::new(Some(Arc::new(schema))),
+            current_catalog: ArcSwap::new(Arc::new(catalog.to_string())),
+            current_schema: ArcSwap::new(Arc::new(schema.to_string())),
         }
     }
 
-    pub fn current_schema(&self) -> Option<String> {
-        self.current_schema.load().as_deref().cloned()
+    pub fn current_schema(&self) -> String {
```
Comment (fengjiachun) [inline src/session/src/context.rs:67]: Why not just return a &str?

File: src/servers/src/http/prometheus.rs [hunk 3]
```diff
@@ -48,10 +51,13 @@ pub async fn remote_write(
 ) -> Result<(StatusCode, ())> {
     let request = decode_remote_write_request(body).await?;
 
-    handler
-        .write(params.db.as_deref().unwrap_or(DEFAULT_SCHEMA_NAME), request)
-        .await?;
+    let ctx = if let Some(db) = params.db {
+        Arc::new(QueryContext::with(DEFAULT_CATALOG_NAME, &db))
```
Comment (fengjiachun) [inline src/servers/src/http/prometheus.rs:55]: prom always use the default catalog?

File: src/session/src/context.rs (hunk 2 above)
Comment (MichaelScofield) [reply src/session/src/context.rs:67]: `current_schema` is guarded by ArcSwap, hard to return a ref into it

File: src/servers/src/http/prometheus.rs (hunk 3 above)
Comment (MichaelScofield) [reply src/servers/src/http/prometheus.rs:55]: yes, catalog is not set as of right now, it's somehow related to cloud or tenant

**Review by fengjiachun — APPROVED** [review]

**Review by killme2008 — APPROVED** [review]
LGTM

File: tests/cases/standalone/alter/rename_table.sql [hunk 4]
```diff
@@ -1,3 +1,7 @@
+CREATE SCHEMA test_rename_table;
```
Comment (killme2008) [inline tests/cases/standalone/alter/rename_table.sql:1]: Why always create a new schema for these tests? I think testing `use` in `catalog` is enough.

**Pull request merged**